WordPress/Classes/Networking/MediaServiceRemoteXMLRPC.m

```diff
@@ -40,6 +40,32 @@ - (void)getMediaWithID:(NSNumber *)mediaID
                  }];
 }
 
+/** Adds a basic auth header to a request if a credential is stored for that specific host.
+ 
+ The credentials will only be added if a set of credentials for the request host are stored on the shared credential storage
+ @param request, the request to where the authentication information will be added.
+ */
+- (void)addBasicAuthCredentialsIfAvailableToRequest:(NSMutableURLRequest *)request
+{
+    NSURLCredentialStorage *credentialStorage = [NSURLCredentialStorage sharedCredentialStorage];
+    NSInteger port = [[request.URL port] integerValue];
+    if (port == 0) {
+        port = 80;
+    }
+    NSURLProtectionSpace *protectionSpace = [[NSURLProtectionSpace alloc] initWithHost:[request.URL host]
+                                                                                  port:port
+                                                                              protocol:[request.URL scheme]
+                                                                                 realm:@"Restricted Area"
```

Comment:
You could probably replace the code in `addBasicAuthCredentialsIfAvailableToRequest` with something like:

```Objective-C
[[[NSURLCredentialStorage sharedCredentialStorage] allCredentials] enumerateKeysAndObjectsUsingBlock:^(NSURLProtectionSpace *ps, NSDictionary *dict, BOOL *stop) {
        [dict enumerateKeysAndObjectsUsingBlock:^(id key, NSURLCredential *credential, BOOL *stop) {
            if ([[ps host] isEqualToString:[request URL].host]) {
                NSString *authStr = [NSString stringWithFormat:@"%@:%@", [credential user], [credential password]];
                NSData *authData = [authStr dataUsingEncoding:NSUTF8StringEncoding];
                NSString *authValue = [NSString stringWithFormat:@"Basic %@", [authData base64EncodedStringWithOptions:NSDataBase64Encoding64CharacterLineLength]];
                [request setValue:authValue forHTTPHeaderField:@"Authorization"];
                *stop = YES;
            }
        }];
    }];
```